tests/lib/nrf_modem_lib/nrf_modem_lib_trace/src/main.c

```diff
@@ -63,15 +62,48 @@ void nrf_modem_lib_trace_callback(enum nrf_modem_lib_trace_event evt)
 	callback_evt = evt;
 }
 
+static int nrf_modem_at_printf_trace_level_valid(const char *fmt, va_list args)
+{
+	TEST_ASSERT_EQUAL_STRING(TRACE_LEVEL_FMT, fmt);
+
+	int val = va_arg(args, int);
+
+	TEST_ASSERT_TRUE(
```

Comment:
We should be explicit here to test that the level is actually what we expect, e.g. `NRF_MODEM_LIB_TRACE_LEVEL_FULL`, not just one of the allowed ones.